sdk/eventhub/event-hubs/src/eventHubConsumerClient.ts

```diff
@@ -117,46 +97,40 @@ export class EventHubConsumerClient {
   constructor(
     host: string,
     eventHubName: string,
-    consumerGroupName: string,
     credential: TokenCredential,
     options?: EventHubClientOptions
-  );
+  );    // #3
   constructor(
-    param1: string,
-    param2: string,
-    param3?: string | EventHubClientOptions,
-    param4?: TokenCredential | EventHubClientOptions,
-    options?: EventHubClientOptions
+    hostOrConnectionString1: string,
+    eventHubNameOrOptions2?: string | EventHubClientOptions,
+    credentialOrOptions3?: TokenCredential | EventHubClientOptions,
+    options4?: EventHubClientOptions
   ) {
-    this._consumersMap = new Map();
-    if (isTokenCredential(param4)) {
-      this._client = new EventHubClient(param1, param2, param4, options);
-      this._consumerGroupName = param3 as string;
-    } else if (typeof param3 === "string") {
-      this._client = new EventHubClient(param1, param2, param4);
-      this._consumerGroupName = param3;
-    } else {
-      this._client = new EventHubClient(param1, param3);
-      this._consumerGroupName = param2;
-    }
-  }
+    if (isTokenCredential(credentialOrOptions3)) {
+      // #3
+      log.consumerClient("Creating client with TokenCredential");
 
-  receive(
-    onMessage: OnMessage,
-    onError: OnError,
-    partitionId: string,
-    abortSignal?: AbortSignalLike
-  ): ReceiveHandler {
-    let consumer = this._consumersMap.get(partitionId);
-    if (!consumer) {
-      consumer = this._client.createConsumer(
-        this._consumerGroupName,
-        partitionId,
-        EventPosition.latest()
+      this._eventHubClient = new EventHubClient(
+        hostOrConnectionString1,
+        eventHubNameOrOptions2 as string,
+        credentialOrOptions3 as TokenCredential,
```

Comment:
Since `isTokenCredential` is a type guard, isn't this parameter already inferred as a TokenCredential?

Reply:
Yes. Fixed the string one below as well!